src/reducers/users.js

```diff
@@ -12,18 +14,35 @@ export default (
         list: [].concat(action.payload.data),
         listLoaded: true,
       });
-    case "USERS_ADD_SUCCEEDED":
-      const existing = state.list.findIndex(
-        (u) => u.id === action.payload.data.userId
-      );
-      if (existing > -1) {
-        return state;
-      }
-      action.payload.data.email = action.payload.data.username;
+    case "PENDING_USERS_LOAD_SUCCEEDED":
       return Object.assign({}, state, {
-        list: [].concat(state.list).concat([action.payload.data]),
-        listLoaded: true,
+        invitationsList: [].concat(action.payload.data.pendingInvitations),
+        invitationsLoaded: true,
       });
+    case "USERS_ADD_SUCCEEDED":
+      console.log(action)
+      if(action.payload.status === 201) {
+        const existing = state.list.findIndex(
+          (u) => u.id === action.payload.data.userId
+        );
+        if (existing > -1) {
+          return state;
+        }
+        action.payload.data.email = action.payload.data.username;
+        return Object.assign({}, state, {
+          list: [].concat(state.list).concat([action.payload.data]),
+          listLoaded: true,
+        });
+      } else if (action.payload.status === 202) {
+        // This means user was added to pending invitations.
```

Comment:
The former solution used an optimistic ui design pattern instead (by adding the new invitation without reloading data from server). It was IMO good solution. Wouldn't you do it the same way?